### ISSUE TITLE
allow npm audit Severity low|high|moderate|critical

### ISSUE THREAD

**Issue by omkadiri** [npm/npm#20596]
I think it will be helpful to be able to audit by severity. Given that some package can have vulnerabilities is thousands.

**Comment by fdenzer** [comment]
As a workaround,

```sh
npm audit | grep High -B3 -A10
```

did the job for me.
Replace High with Low/Moderate/Critical as you see fit.

**Comment by joepie91** [comment]
Having lots of known-vulnerability warnings shouldn't be a reason to hide them - it just means you have a lot of unresolved issues that need to be fixed.

Issues of *any* severity should really be fixed, the severity of a vulnerability assessment should only serve to prioritize the fixes; for example, it would be reasonable to list vulnerabilities with the highest severity first or last, if that's not already being done.

**Comment by byCedric** [comment]
Is this similar to https://github.com/npm/npm/issues/20558?

**Comment by bendykowski** [comment]
@byCedric No, it's not. #20558 is about introducing flags for better exit code management.

@joepie91 Grouping vulnerabilities by severity is not a bad idea. However, the ability to filter the output by severity would be a great addition as well. Having hundreds low priority vulnerabilities can introduce mess and be a pain when looking for the most critical ones. If you create a project from scratch you can take care of all of them part by part. It's much harder for existing projects someone forgot to take care. The best option to start maintenance and fortifying your software are baby steps, starting from the most critical issues. The ability to filter them help you focus step by step on each level. It doesn't mean low severity vulnerabilities are fine, of course not, they are just not that important to start with. It should be just a step in the whole maintenance process and not a permanent conscience mute of course.

**Comment by quinnlangille** [comment]
Yeah, I'm trying to use the audit feature as a blocker for a CI-pipeline and @fdenzer's solution seems to work exactly how I'd expect a native flag to work. I did expand on it a bit by adding a small regexp to filter for multiple levels:

`npm audit | grep -E "(High | Critical)" -B3 -A10 `

Would still be nice to see some more options for `npm audit` though, it's a great tool!

**Comment by welwood08** [comment]
Just a head's up since I'd forgotten I was subscribed here, npm/npm-audit-report#13 is a prerequisite of this and I've submitted a PR to implement it. Once that's sorted, adding flags to the CLI should be no problem.
